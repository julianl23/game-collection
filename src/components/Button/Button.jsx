import React from "react";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import { switchProp } from "styled-tools";

const sizes = {
  small: 25,
  normal: 36,
  large: 50,
};

const buttonStyles = {
  primary: "#322F6F",
  secondary: "#F7FAFE",
};

const textStyles = {
  primary: "#fff",
  secondary: "#222629",
};

const StyledButton = styled.button`
  height: ${({ size }) => sizes[size]}px;
  background: ${({ buttonStyle }) => buttonStyles[buttonStyle]};
  border-radius: 3px;
  font-family: ${({ theme }) => theme.fontFamily};
  font-size: 14px;
  font-weight: 300;
  line-height: 16px;
  color: ${({ buttonStyle }) => textStyles[buttonStyle]};
  cursor: pointer;
  ${switchProp("block", {
    true: css`
      display: block;
      width: 100%;
    `,
    false: css`
      display: inline-block;
    `,
  })};
`;

const Button = ({
  children,
  type,
  className,
  buttonStyle,
  size,
  onClick,
  block,
}) => (
  <StyledButton
    className={className}
    type={type}
    buttonStyle={buttonStyle}
    size={size}
    onClick={onClick}
    block={block}
  >
    {children}
  </StyledButton>
);

Button.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  type: PropTypes.oneOf(["button", "submit", "reset"]),
  buttonStyle: PropTypes.oneOf(["primary", "secondary"]),
  size: PropTypes.oneOf(["small", "normal", "large"]),
  onClick: PropTypes.func,
  block: PropTypes.bool,
};

Button.defaultProps = {
  children: "",
  className: "",
  type: "button",
  buttonStyle: "primary",
  size: "normal",
  onClick: null,
  block: false,
};

export default Button;
