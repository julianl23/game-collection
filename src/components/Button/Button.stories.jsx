import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import { withKnobs, select, text, boolean } from "@storybook/addon-knobs";
import { jsxDecorator } from "storybook-addon-jsx";

import Button from "./Button";

const types = ["button", "submit", "reset"];
const styles = ["primary", "secondary", "turquoise"];
const sizes = ["small", "normal", "large"];

const stories = storiesOf("Button", module);
stories.addDecorator(withKnobs);
stories.addDecorator(jsxDecorator);

stories.add("with knobs", () => (
  <Button
    onClick={action("Clicked Button")}
    type={select("Type", types, "button")}
    buttonStyle={select("Button Style", styles, "primary")}
    size={select("Size", sizes, "normal")}
    block={boolean("Block", false)}
  >
    {text("Button Text", "Hello Button")}
  </Button>
));
