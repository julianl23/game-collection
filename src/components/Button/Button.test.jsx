import React from "react";
import TestRenderer from "react-test-renderer";
import Button from "./Button";

describe("Button", () => {
  it("renders correctly", () => {
    const instance = TestRenderer.create(<Button>Hello World</Button>);
    expect(instance.toJSON()).toMatchSnapshot();
  });
});
