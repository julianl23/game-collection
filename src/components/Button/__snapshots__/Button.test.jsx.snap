// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  height: 36px;
  background: #322F6F;
  color: #fff;
}

<button
  className="c0"
  size="normal"
  type="button"
>
  Hello World
</button>
`;
